documentation/en/api-experimental-user-facing-v2-docs/api-v2-experimental.md

```diff
@@ -0,0 +1,1219 @@
+
+## Consistency Across Selector Usage
+
+### Avoiding Inconsistent State Views
+
+When working with blockchain data, particularly for operations involving multiple state reads (like checking balances across accounts or calculating totals), it's crucial to maintain consistency in how you select the reference tipset. Using different selectors (`latest` vs `safe` vs `finalized`) or querying at different points in time for related data can lead to inconsistent views of the state and incorrect results.
+
+### Risk of Double Counting or Missing State Changes
+
+Consider this scenario where inconsistent selector usage leads to double counting of funds:
+
+1. A user has two accounts:
+    - Account1 initially contains 5 FIL
+    - Account2 initially contains 0 FIL
+2. A transaction is created that transfers 5 FIL from Account1 to Account2
+3. The transaction is included in the latest tipset but has not yet been finalised
+4. Application queries Account1 using the `finalized` selector at Time T1
+    - This returns 5 FIL (because at the finalised tipset, the transfer hasn't happened yet)
+5. A new finality certificate is produced, finalising the transfer transaction
+6. Application queries Account2 using the `finalized` selector at Time T2 (after new finalisation)
+    - This returns 5 FIL (because now the transfer is finalised)
+7. If the application adds these balances, it would incorrectly conclude the user has 10 FIL total, when they actually have 5 FIL
+
+To avoid such inconsistencies:
+
+1. **Lock to a Specific TipSet Key (Recommended for High Consistency):**
+    - First, retrieve a specific TipSet using a chosen selector (e.g., `finalized`, `safe`, or a specific `height`).
+    - Extract its `TipSetKey` (the array of CIDs).
+    - Use *that same key* via `TipSetSelectors.Key(retrievedKey)` for *all* related state queries (e.g., `StateGetActor`, `StateGetID`) within that logical operation.
+    - This ensures all queries operate on the exact same snapshot of the blockchain state.
+
+2. **Use the Same Tag Consistently:**
+    - If absolute consistency isn't required but related data should be viewed from the *same level of finality*, consistently use the *same tag* (e.g., always use `safe` or always use `finalized`) for all related queries within an operation.
+    - *Caution*: Even using the same tag across multiple sequential calls might hit slightly different tipsets if the chain advances between calls. Locking to a key is stricter.
+
+3. **Query at a Specific Height (Use with Care):**
+    - Choose a specific height.
+    - Use `TipSetSelectors.Height(chosenHeight, previous, anchor)` consistently for all related queries.
+    - *Caution*: Ensure the `anchor` and `previous` flag are also used consistently. Using `previous: true` might result in slightly different tipsets if the exact height is null. Using `previous: false` provides a stricter height match but might fail if the height is null.
```

Comment:
I don't think recommendations 2 and 3 should be here at all

Reply:
Ack, good point.  I have cleaned this up in https://github.com/filecoin-project/lotus/pull/13051/commits/bbd32817386a869a0c0df367e7fb75fb3ba1059b .  Resolving.